lib/xandra/protocol.ex

```diff
@@ -18,6 +18,12 @@ defmodule Xandra.Protocol do
     %{frame | body: encode_string_map(requested_options)}
   end
 
+  def encode_request(%Frame{kind: :auth_response} = frame, _requested_options, options) do
+    {authenticator, auth_options} = Keyword.fetch!(options, :authentication)
```

Comment:
yeah, Cassandra will send an authenticate challenge if there's a need of it, which depends on the authentication type user configured. Together with `Authenticator` behavior, it makes authentication totally extendable by users.

> should we raise a better error such as "C* wants to auth, but you didn't provide :authentication" or something like that?

Personally I think a clear failure message sounds very good 👍